lab_3_ann_retriever/main.py

```diff
@@ -200,26 +248,32 @@ def load(self, file_path: str) -> bool:
         Save the Vectorizer state to file.
 
         Args:
-            file_path (str): The path to the file from which to load the instance
+            file_path (str): The path to the file where the instance will be saved.
 
         Returns:
-            bool: True if the vectorizer was saved successfully
+            bool: True if the vectorizer saved successfully.
 
         In case of corrupt input arguments, False is returned.
         """
 
-    def _calculate_tf_idf(self, document: list[str]) -> Vector | None:
-        """
-        Get TF-IDF for document.
 
-        Args:
-            document (list[str]): Tokenized document to vectorize
+def calculate_distance(query_vector: Vector, document_vector: Vector) -> float | None:
+    """
+    Calculate Euclidean distance for a document vector.
 
-        Returns:
-            Vector | None: TF-IDF vector for document
+    Args:
+        query_vector (Vector): Vectorized vector.
+        document_vector (Vector): Vectorized documents.
 
-        In case of corrupt input arguments, None is returned.
-        """
+    Returns:
+        float | None: Euclidean distance for vector.
+
+    In case of corrupt input arguments, None is returned.
+    """
+    if query_vector is None or document_vector is None:
+        return None
+
+    return sum((query - doc) ** 2 for query, doc in zip(query_vector, document_vector)) ** 0.5
```

Comment:
why 0.5?

Reply:
so that i can take the square root of the sum